src/zmq/zmqrpc.cpp

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) 2018 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <zmq/zmqrpc.h>
+
+#include <rpc/server.h>
+#include <zmq/zmqabstractnotifier.h>
+#include <zmq/zmqnotificationinterface.h>
+
+#include <univalue.h>
+
+namespace {
+
+UniValue getzmqnotifications(const JSONRPCRequest& request)
+{
+    if (request.fHelp || request.params.size() != 0)
+        throw std::runtime_error(
+            "getzmqnotifications\n"
+            "\nReturns information about the active ZeroMQ notifications.\n"
+            "\nResult:\n"
+            "{\n"
+            "  \"publish\": [               (json array) Active publishers\n"
+            "    {                        (json object)\n"
+            "      \"type\": \"pubhashtx\",   (string) Type of notification\n"
+            "      \"address\": \"...\"       (string) Address of the publisher\n"
+            "    },\n"
+            "    ...\n"
+            "  ]\n"
+            "}\n"
+            "\nExamples:\n"
+            + HelpExampleCli("getzmqnotifications", "")
+            + HelpExampleRpc("getzmqnotifications", "")
+        );
+
+    UniValue publish(UniValue::VARR);
+    if (g_zmq_notification_interface != nullptr) {
+        const std::list<const CZMQAbstractNotifier*> notifiers = g_zmq_notification_interface->GetActiveNotifiers();
+        for (const auto* n : notifiers) {
+            UniValue obj(UniValue::VOBJ);
+            obj.pushKV("type", n->GetType());
+            obj.pushKV("address", n->GetAddress());
+            publish.push_back(obj);
+        }
+    }
+
+    UniValue result(UniValue::VOBJ);
+    result.pushKV("publish", publish);
```

Comment:
Now that each notification has the `type`, could just return the array:
```
[
    { "type": "...", "address": "..." },
    ...
]
```

Reply:
Makes sense, changed.